xsd-parser-rs/src/parser/tests.rs

```diff
@@ -63,4 +63,65 @@ mod test {
             _ => unreachable!(),
         }
     }
+
+    #[test]
+    fn test_restriction_any_type() {
+        use crate::parser::parse;
+        use crate::parser::types::RsEntity;
+
+        let text = r#"
+<xs:schema targetNamespace="http://schemas.xmlsoap.org/ws/2005/04/discovery"
+    xmlns:tns="http://schemas.xmlsoap.org/ws/2005/04/discovery"
+    xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema"
+    elementFormDefault="qualified" blockDefault="\#all">
+
+    <xs:element name="AppSequence" type="tns:AppSequenceType"/>
+    <xs:complexType name="AppSequenceType">
+        <xs:complexContent>
+            <xs:restriction base="xs:anyType">
+                <xs:attribute name="InstanceId" type="xs:unsignedInt" use="required"/>
+                <xs:attribute name="SequenceId" type="xs:anyURI"/>
+                <xs:attribute name="MessageNumber" type="xs:unsignedInt" use="required"/>
+                <xs:anyAttribute namespace="\#\#other" processContents="lax"/>
+            </xs:restriction>
+        </xs:complexContent>
+    </xs:complexType>
+
+</xs:schema>
+        "#;
+
+        let result = parse(text).unwrap();
+        println!("{:?}", result); // TODO: remove
```

Comment:
It is not necessary to remove printlns from tests because it will actually print only when test fails, otherwise printlns are ignored.

Reply:
Good point